cmd/tether/main_linux.go

```diff
@@ -33,22 +33,17 @@ import (
 var tthr tether.Tether
 
 func init() {
-	sigs := make(chan os.Signal, 1)
-	signal.Notify(sigs, syscall.SIGHUP)
-
 	// Initiliaze logger with default TextFormatter
 	log.SetFormatter(viclog.NewTextFormatter())
 
 	// use the same logger for trace and other logging
 	trace.Logger = log.StandardLogger()
 	log.SetLevel(log.DebugLevel)
 
-	go func() {
-		for _ = range sigs {
-			log.Infof("Got A HUP signal! Reloading tether...")
-			tthr.Reload()
-		}
-	}()
+	// init and start the signal handler
+	if err := startSignalHandler(); err != nil {
```

Comment:
why would you even want to check for error if function always return nil? I would just update that function to return nothing, so less lines of code.